enhancements/local-storage/subprovisioner-integration-into-lvms.md

```diff
@@ -0,0 +1,361 @@
+  - Provide rollback mechanisms and detailed instructions to revert to previous versions. Ensure that downgrades do not result in data loss or service interruptions.
+    The operator should ensure that the shared VGs can be cleaned up manually.
+  - Ensure that downgrades do not result in data loss or service interruptions. The operator should ensure that the shared VGs can be cleaned up without data loss on other device classes.
+
+### Version Skew Strategy
+
+- Ensure compatibility between different versions of LVMS and the integrated Subprovisioner CSI driver.
+  - Implement version checks and compatibility checks in the `vgmanager` component.
+  - Ensure that the operator can handle version skew between the LVMS operator and the Subprovisioner CSI driver where required.
+  - Provide clear guidelines on how to manage version skew and perform upgrades in a controlled manner.
+  - One version of LVMS should be able to handle one version of the Subprovisioner CSI driver.
+- Document supported version combinations and any known issues with version mismatches.
+- Provide clear guidelines on how to manage version skew and perform upgrades in a controlled manner.
+
+### Security Considerations
+
+- **Access Control**:
+  - Ensure that access to shared storage is controlled and restricted to authorized users. Node-level access control should be enforced similarly to TopoLVM.
+  - Implement RBAC policies to restrict access to VGs and CSI drivers based on user roles and permissions.
+  - Ensure that shared VGs are only accessible by nodes that are authorized to access them.
+- **CVE Scanning**:
+  - Ensure that the Subprovisioner CSI driver is regularly scanned for vulnerabilities and that any identified issues are addressed promptly.
+  - Implement a process for CVE scanning and remediation for the Subprovisioner CSI driver.
+  - Fixes for CVEs should be handled in a dedicated midstream openshift/subprovisioner for critical CVEs when Red Hat decides to no longer solely own the project. Until then, the fixes will be handled by the Red Hat team and a midstream is optional.
+
+### Implementation Milestones
+
+- **Phase 1**: Initial design and prototyping. Basic integration with Subprovisioner and updates to the LVMCluster CR.
+- **Phase 2**: Development of `vgmanager` functionalities for VG orchestration and management. Integration and E2E testing.
+- **Phase 3**: Performance testing, bug fixes, and documentation. Preparing for Alpha release.
+- **Phase 4**: Developer Preview release with comprehensive manual and QE testing. Gathering user feedback and making improvements.
+- **Phase 5**: Technology PReview with Documentation Extension and preparation of GA.
+- **Phase 6**: General Availability (GA) release with proven stability and performance in production environments.
+
+### Drawbacks
```

Comment:
Are there any drawbacks here in the context of the original intent of LVMS being targeted at edge use cases?

For example, how much impact would we expect to the runtime footprint of the operator running TopoLVM + Subprovisioner together or is it essentially a no-op for current LVMS users only using TopoLVM based storage?

Reply:
The runtime footprint impact should be exactly one switch case statement that determines which driver to instantiate (what you correctly assume is basically a no-op). If someone does not make use of Subprovisioner, then nothing for it will be created / instantiated.

There will be an impact on the footprint of the container images and also the runtime of a subprovisioner instance will likely be at least 10x higher due to us having to run sanlock and lvmlockd (if using sanlock as is expected currently in subprovisioner), but that should not concern edge users.